src/core/Directus/Services/RolesService.php

```diff
@@ -157,7 +158,7 @@ public function canDelete($id, $fetchNew = false)
         }
 
         // TODO: RowGateWay should parse values against their column type
-        return !(!$group || $group->id == 1 || strtolower($group->name) === 'public');
+        return !(!$group || $group->id == 1 || strtolower($group->name) === Roles::PUBLIC);
     }
```

Comment:
Same thing here, before it seemed to be checking that `$group->name === 'public'` now it's checking that `$group->name === 2`, right? 

I would imagine something like `Roles::getName(Roles::PUBLIC)` — Not sure on that code but, that's the idea haha, or instead of using the name at all, we could just change the logic to use group id instead.

Reply:
Agree with you @shealavington 

If I am right then this condition was there to make sure that the user should not delete `Public` or `Admin` role. Earlier user can remove `Public` and `Admin` role using `API`. So this code was not worked as expected.

Reverted this PR and sending another one which adds this functionality.